app/services/foreman/unattended_installation/host_finder.rb

```diff
@@ -54,7 +54,12 @@ def find_host_by_ip_or_mac
 
         mac_list = query_params[:mac_list]
 
-        query = mac_list.empty? ? { :nics => { :ip => ip } } : ["lower(nics.mac) IN (?)", mac_list]
+        if address_parser.ipv6?
+          query = mac_list.empty? ? { :nics => { :ip6 => ip } } : ["lower(nics.mac) IN (?)", mac_list]
+        else
+          query = mac_list.empty? ? { :nics => { :ip => ip } } : ["lower(nics.mac) IN (?)", mac_list]
+        end
```

Comment:
This removes duplication of logic, which I think is important.
```suggestion
        query = mac_list.empty? ? { :nics => { (address_parser.ipv6? ? :ip6 : :ip) => ip } } : ["lower(nics.mac) IN (?)", mac_list]
```

Though it's a bit unreadable. So I think this is better:

```suggestion
        query = if mac_list.empty?
            column = address_parser.ipv6? ? :ip6 : :ip
            { :nics => { column => ip } }
          else
            ["lower(nics.mac) IN (?)", mac_list]
          end
```

Reply:
Thanks for the suggested improvement, I am not a ruby developer. Tested against our installation and it works. Will push a new commit.